auth/store_test.go

```diff
@@ -230,6 +231,126 @@ func TestRoleAdd(t *testing.T) {
 	}
 }
 
+func setupAuthStore(t *testing.T) (AuthStore, func(t *testing.T)) {
+	b, tPath := backend.NewDefaultTmpBackend()
+
+	as := NewAuthStore(b, dummyIndexWaiter)
+	roleAddErr := enableAuthAndCreateRoot(as)
+	if roleAddErr != nil {
+		t.Fatal(roleAddErr)
+	}
+
+	// adds a new role
+	_, roleAddErr = as.RoleAdd(&pb.AuthRoleAddRequest{Name: "role-test"})
+	if roleAddErr != nil {
+		t.Fatal(roleAddErr)
+	}
+	as.UserAdd(&pb.AuthUserAddRequest{Name: "user1", Password: "xyz"})
+	as.UserGrantRole(&pb.AuthUserGrantRoleRequest{User: "user1", Role: "role-test"})
+
+	return as, func(t *testing.T) {
+		func() {
+			b.Close()
+			os.Remove(tPath)
+		}()
+		as.Close()
+	}
+}
+
+func TestGetUser(t *testing.T) {
+	as, tearDown := setupAuthStore(t)
+	defer tearDown(t)
+	resp, userGetErr := as.UserGet(&pb.AuthUserGetRequest{Name: "user1"})
+	if userGetErr != nil {
+		t.Fatal(userGetErr)
+	}
+	assert.Equal(t, []string{"role-test"}, resp.Roles, "It should not have any roles")
```

Comment:
Sure, I can change it if..else. I have seen it used in ./store/state_test.go. Just curious for my knowledge, why it is not recommended practice?